File: parking_app-main-main/API/SmartParkingAPI/Models/Entities/Section.cs
using System.ComponentModel.DataAnnotations.Schema;
namespace SmartParkingAPI.Models.Entities

{
    public class Section
    {
        public string SectionId { get; set; }
        public int GarageId { get; set; }
        public int TotalSpots { get; set; }
        public int Available { get; set; }

        // ✅ Add this if EF expects navigation between Section → Garage
        [ForeignKey("GarageId")]
        public Garage Garage { get; set; }  // <-- this resolves the error
    }
}

File: parking_app-main-main/lib/models/section.dart
class Section {
  final String sectionId;
  final int garageId;
  final int totalSpots;
  final int available; // stored as int (e.g., 1 or 0)

  Section({
    required this.sectionId,
    required this.garageId,
    required this.totalSpots,
    required this.available,
  });

  factory Section.fromJson(Map<String, dynamic> json) {
    return Section(
      sectionId: json['sectionId'] ?? '',
      garageId: json['garageId'] ?? 0,
      totalSpots: json['totalSpots'] ?? 0,
      available: json['available'] ?? 0,
    );
  }
}
